### PR TITLE
Remove SamplingHint from the Sampling otep

### PR DESCRIPTION
**Pull request by bogdandrutu**
The main motivation is that we did not proved the necessity of the SamplinHint well enough, and we can always add this in the future (applying the principal of a minimal API surface).

### CODE REVIEW

**Comment by bogdandrutu** [comment]
As an author I think it is easier to modify that otep, and also is not a change that was documented in the specs.

**Comment by bogdandrutu** [comment]
@open-telemetry/specs-approvers please review, we need couple more reviews to merge this

File: text/0006-sampling.md [hunk 1]
```diff
@@ -68,9 +67,10 @@ Examples: gRPC, Express, Django developers.
 
 **Solution:**
 
- * On the start Span operation, the OpenTelemetry API will allow marking a span with one of three
- choices for the [SamplingHint](#samplinghint).
- 
+* For the moment opentelemetry API will not offer any `SamplingHint` functionality for the last use case,
```
Comment (jkwatson) [inline text/0006-sampling.md]:
```suggestion
* For the moment the OpenTelemetry API will not offer any `SamplingHint` functionality for the last use case,
```

File: text/0006-sampling.md [hunk 2]
```diff
@@ -68,9 +67,10 @@ Examples: gRPC, Express, Django developers.
...
+this is intentional to avoid premature optimizations, and it is based on the fact that changing an API is
```
Comment (jkwatson) [inline text/0006-sampling.md]:
```suggestion
This is intentional to avoid premature optimizations, and it is based on the fact that changing an API is
```

**Pull request merged**